backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py

```diff
@@ -255,8 +257,14 @@ def list_shared_databases_tables_with_env_group(environmentUri: str, groupUri: s
 
     @staticmethod
     def resolve_shared_db_name(GlueDatabaseName: str, shareUri: str, targetEnvAwsAccountId: str, targetEnvRegion: str):
-        old_shared_db_name = (GlueDatabaseName + '_shared_' + shareUri)[:254]
-        database = GlueClient(
-            account_id=targetEnvAwsAccountId, database=old_shared_db_name, region=targetEnvRegion
-        ).get_glue_database()
-        return old_shared_db_name if database else GlueDatabaseName + '_shared'
+        with get_context().db_engine.scoped_session() as session:
+            share = ShareObjectRepository.get_share_by_uri(session, shareUri)
+            dataset = DatasetBaseRepository.get_dataset_by_uri(session, share.datasetUri)
+            datasetGlueDatabase = GlueClient(
+                account_id=dataset.AwsAccountId, region=dataset.region, database=GlueDatabaseName
+            ).get_glue_database_from_catalog()
```

Comment:
Can we wrap this in `try/except` - just in case if `get_glue_database_from_catalog` raises an error for 1 DB then it will impact loading all of the share DB items in worksheet

other than that I think this PR is good to go!

Reply:
Thanks for spotting that. Added Try catch